src/main/webapp/app/overview/submission-result-status.component.html

```diff
@@ -1,11 +1,11 @@
 <div class="col-auto">
-    <ng-container *ngIf="studentParticipation && studentParticipation.results?.length; else noResult">
+    <ng-container *ngIf="shouldShowResult; else noResult">
```

Comment:
```suggestion
    <ng-container *ngIf="shouldShowResult; else showNoResult">
```
Conditional is more intuitive then.